### PR TITLE
Update the documentation for the new variant normalizer

### PR DESCRIPTION
**Pull request by kianfar77**
## What changes are proposed in this pull request?
This PR updates the documentation for the new variant normalizer.

## How is this patch tested?
- [ ] Unit tests
- [x] Integration tests
- [x] Manual tests

(Details)

### CODE REVIEW

File: docs/source/etl/variant-normalization.rst [hunk 1]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
-     - Path to the reference genome ``.fasta`` or ``.fa`` file (required for normalization)
-
-       **Note**: ``.fai`` and ``.dict`` files with the same name must be present in the same folder.
-
+     - Path to the reference genome ``.fasta`` or ``.fa`` file. This file must be accompanied with a .fai index file in the same folder.
+   * - ``replace_columns``
+     - boolean
+     - |
+       | ``False``: The transformer does not touch the original ``start``, ``end``, ``referenceAllele`` and ``alternateAlleles`` columns. Instead, a StructType column called ``normalizationResult`` is added to the DataFrame which contains the normalized ``start``, ``end``, ``referenceAllele``, and ``alternateAlleles`` columns as well as the ``normalizationStatus`` StructType as the fifth field, which contains the following subfields:
+       |        ``changed``: Indicates whether the variant data was changed as a result of normalization or not.
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: remove `or not`

File: docs/source/etl/variant-normalization.rst [hunk 2]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
+       |        ``errorMessage``: An error message in case the attempt at normalizing the row hit an error. In this case, the ``changed`` field will be set to ``false``. If no errors occur this field will be ``null``. In case of error, the first four fields in ``normalizationResult`` will be ``null``.
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: We should be consistent with whether boolean values are capitalized (Scala vs Python style).

File: docs/source/etl/variant-normalization.rst [hunk 3]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
+       |
+       | ``True`` (default): The original ``start``, ``end``, ``referenceAllele``, and ``alternateAlleles`` columns are replaced with the normalized values in case they have changed. Otherwise (in case of no change or an error), the original ``start``, ``end``, ``referenceAllele``, and ``alternateAlleles`` are not touched. A StructType ``normalizationStatus`` column is added to the DataFrame with the same subfields explained above.
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: `touched` -> `modified`

File: docs/source/etl/variant-normalization.rst [hunk 4]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
-     - | ``normalize``: Only normalizes the variants (if user does not pass the option, ``normalize`` is assumed as default)
+     - |
+       | ``normalize``: Only normalizes the variants (if user does not pass the option, ``normalize`` is assumed as default)
        | ``split_and_normalize``: Split multiallelic variants to biallelic variants and then normalize the variants. This usage is deprecated. Instead, use :ref:`split_multiallelics transformer<split_multiallelics>` followed by normalize_variants transformer.
        | ``split``: Only split the multiallelic variants to biallelic without normalizing. This usage is deprecated. Instead, use :ref:`split_multiallelics transformer<split_multiallelics>`.
 
-.. notebook:: .. etl/normalizevariants-transformer.html
+``normalize_variant`` Function
+==============================
+The normalizer can also be used as a SQL expression function applied on proper DataFrame columns to get normalization results as a new column. See the discreption of this function at :ref:`Glow PySpark Functions<pyspark_functions>` for details on how to use it in the Python API. The function can be similarly used in Scala API or as a SQL expression.
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: Replace this section with: The normalizer can also be used as a SQL expression function. See :ref:`Glow PySpark Functions<pyspark_functions>` for details on how to use it in the Python API.

Comment (karenfeng) [reply docs/source/etl/variant-normalization.rst]: We haven't explained how to use Scala API yet, so we should put off mentioning it for now. We can, however, create a code block demonstrating how to use the SQL expression.

File: docs/source/etl/variant-normalization.rst [hunk 5]
```diff
@@ -13,18 +13,24 @@ Variant Normalization
...
+  Glow's variant normalization algorithm follows the same logic as those used in normalization tools such as `bcftools norm <https://www.htslib.org/doc/bcftools.html#norm>`_ and `vt normalize <https://genome.sph.umich.edu/wiki/Vt#Normalization>`_. This normalization logic is different from the one used by GATK's `LeftAlignAndTrimVariants <https://gatk.broadinstitute.org/hc/en-us/articles/360037225872-LeftAlignAndTrimVariants>`_, which sometimes yields incorrect normalization (see `Variant Normalization <https://genome.sph.umich.edu/wiki/Variant_Normalization>`_ for more details).
+
+``normalize_variants`` Transformer
+==================================
+
+``normalize_variants`` transformer can be applied on a variant DataFrame (e.g., generated by loading VCF or BGEN files) to normalize its variants, bringing unprecedented scalability to this operation. The output DataFrame schema can be controlled using the ``replace_columns`` option as explained below.
```
[the remark this answers is not in the document]
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst [hunk 6]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
      - Type
      - Possible values and description
-   * - ``referenceGenomePath``
+   * - ``reference_genome_path``
      - string
-     - Path to the reference genome ``.fasta`` or ``.fa`` file (required for normalization)
-
-       **Note**: ``.fai`` and ``.dict`` files with the same name must be present in the same folder.
-
+     - Path to the reference genome ``.fasta`` or ``.fa`` file. This file must be accompanied with a .fai index file in the same folder.
```
[the remark this answers is not in the document]
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst [hunk 7]
```diff
@@ -57,17 +63,27 @@ The ``normalize_variants`` transformer has the following options:
...
+   * - ``replace_columns``
+     - boolean
+     - |
+       | ``False``: The transformer does not touch the original ``start``, ``end``, ``referenceAllele`` and ``alternateAlleles`` columns. Instead, a StructType column called ``normalizationResult`` is added to the DataFrame which contains the normalized ``start``, ``end``, ``referenceAllele``, and ``alternateAlleles`` columns as well as the ``normalizationStatus`` StructType as the fifth field, which contains the following subfields:
```
[the remark this answers is not in the document]
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 1 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 2 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 3 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 4 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst [hunk 8]
```diff
@@ -13,18 +13,24 @@ Variant Normalization
...
+Different genomic analysis tools often represent the same genomic variant in different ways, making it non-trivial to integrate and compare variants across call sets. Therefore, **variant normalization** is an essential step to be applied on variants before further downstream analysis to make sure the same variant is represented identically in different call sets. Normalization is achieved by making sure the variant is parsimonious and left-aligned (see `Variant Normalization <https://genome.sph.umich.edu/wiki/Variant_Normalization>`_ for more details).
 
-Glow provides the ``normalize_variants`` transformer to be applied on a variant DataFrame to normalize its variants, bringing unprecedented scalability to this operation. When applied on an input DataFrame of variants (e.g., generated by loading VCF or BGEN files), this transformer generates a DataFrame containing normalized variants.
+Glow provides variant normalization capabilities as a DataFrame transformer as well as a SQL expression function with Python and Scala APIs, bringing unprecedented scalability to this operation.
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: Let's leave out the Scala part for now, as we don't have Scala API documentation yet. `Glow provides variant normalization capabilities as a DataFrame transformer as well as a SQL expression function with Python and Scala APIs, bringing unprecedented scalability to this operation.` -> `Glow provides variant normalization capabilities as a DataFrame transformer as well as a SQL expression function with a Python API, bringing unprecedented scalability to this operation.`

File: docs/source/etl/variant-normalization.rst [hunk 9]
```diff
@@ -57,17 +63,33 @@ The ``normalize_variants`` transformer has the following options:
...
-.. notebook:: .. etl/normalizevariants-transformer.html
+``normalize_variant`` Function
+==============================
+The normalizer can also be used as a SQL expression function. See :ref:`Glow PySpark Functions<pyspark_functions>` for details on how to use it in the Python API. An example of the SQL expression usage is demonstrated below:
+
+.. code-block:: python
+
+  from pyspark.sql.functions import expr
+  normalization_expr = "normalize_variant(contigName, start, end, referenceAllele, alternateAlleles, '{ref_genome}')".format(ref_genome=ref_genome_path)
+  df_normalized = df_original.withColumn('normalizationResult', expr(normalization_expr))
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: Please add a test for this with `.. invisible-code-block: python`

File: docs/source/etl/variant-normalization.rst [hunk 10]
```diff
@@ -57,17 +63,33 @@ The ``normalize_variants`` transformer has the following options:
...
-     - | ``normalize``: Only normalizes the variants (if user does not pass the option, ``normalize`` is assumed as default)
+     - |
+       | ``normalize``: Only normalizes the variants (if user does not pass the option, ``normalize`` is assumed as default)
        | ``split_and_normalize``: Split multiallelic variants to biallelic variants and then normalize the variants. This usage is deprecated. Instead, use :ref:`split_multiallelics transformer<split_multiallelics>` followed by normalize_variants transformer.
        | ``split``: Only split the multiallelic variants to biallelic without normalizing. This usage is deprecated. Instead, use :ref:`split_multiallelics transformer<split_multiallelics>`.
 
-.. notebook:: .. etl/normalizevariants-transformer.html
+``normalize_variant`` Function
+==============================
+The normalizer can also be used as a SQL expression function. See :ref:`Glow PySpark Functions<pyspark_functions>` for details on how to use it in the Python API. An example of the SQL expression usage is demonstrated below:
```
Comment (karenfeng) [inline docs/source/etl/variant-normalization.rst]: Replace last sentence with: "An example of an expression using the `normalize_variant` function is below:"

Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 9 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

File: docs/source/etl/variant-normalization.rst (hunk 8 above)
Comment (kianfar77) [reply docs/source/etl/variant-normalization.rst]: Done

**Pull request merged**